### ISSUE TITLE
pyinstaller created executable immediately gives traceback - schwifty related

### ISSUE THREAD

**Issue by Xargas** [pyinstaller/pyinstaller#8458]
I wrote myself an application where I wanted an EXE file for.

I can reproduce the bug with this minimal version of code in a python file:
```
#/bin/python

from schwifty import IBAN, BIC

try:
    IBAN("AT12") # example
except Exception as e:
    print("IBAN: " + str(e))
```

When run with "python minimal.py" it gives the expected result: "IBAN: Invalid IBAN length"

Then I create the executable with:
```
pyinstaller.exe minimal.py -y --onefile
```
This creates the folders build and dist. When I run the dist/minimal.exe file (from cmd.exe), I get:
```
C:\minimal\dist>minimal.exe
Traceback (most recent call last):
  File "minimal.py", line 3, in <module>
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "PyInstaller\loader\pyimod02_importers.py", line 419, in exec_module
  File "schwifty\__init__.py", line 6, in <module>
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "PyInstaller\loader\pyimod02_importers.py", line 419, in exec_module
  File "schwifty\bban.py", line 8, in <module>
  File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
  File "PyInstaller\loader\pyimod02_importers.py", line 419, in exec_module
  File "schwifty\bic.py", line 496, in <module>
  File "schwifty\registry.py", line 93, in build_index
  File "schwifty\registry.py", line 71, in get
ValueError: Failed to load registry bank
[9936] Failed to execute script 'minimal' due to unhandled exception!
```

At first I thought it was related to [https://github.com/pyinstaller/pyinstaller/issues/5390](https://github.com/pyinstaller/pyinstaller/issues/5390), but as I am using a very current version this should not be related from an outside view (I did not look into pyinstaller code in more detail)

Do you have an idea how this could be fixed?

**Comment by rokm** [comment]
It's a two-fold problem.

First, the data files from `schwifty/bank_registry` are not collected. Typically, this would be addressed by adding `--collect-data schwifty` to your PyInstaller command.

However, it looks like `schwifty` is attempting to adjust search path for PyInstaller-frozen application ([here](https://github.com/mdomke/schwifty/blob/99a1dcc8ee3cf55224e0751a8c5eb98c3c637127/schwifty/registry.py#L55-L58)), and does so incorrectly, because the `bank_registry` data directory is (should be) collected as `sys._MEIPASS/schwifty/bank_registry` and not  `sys._MEIPASS/bank_registry` (unless you explicitly tell PyInstaller to collect package contents into top-level directory, as suggested in https://github.com/mdomke/schwifty/issues/196).

The canonical solution here would be for `schwifty` to revert https://github.com/mdomke/schwifty/pull/92, as PyInstaller supports `importlib.resources.files()` since v4.6. That does require collection of metadata as well, so you need `--collect-data schwifty --copy-metadata schwifty` (EDIT: looks like metadata is required even if `importlib.resources.files()` is not used).

**Comment by rokm** [comment]
We'll add a hook for `schwifty` to `pyinstaller-hooks-contrib` once the upstream removes the problematic search path modification.

**Comment by mdomke** [comment]
@rokm The problem should be fixed with `schwifty` releaese `2024.05.2`

**Comment by Xargas** [comment]
@rokm Thanks for the explanation! It was the first time I used pyinstaller. I would have needed much longer to get this deep into it!
@mdomke This version hint with the info from @rokm was the info I needed! Thanks!

I now used (updated using pip):
- pyinstaller 6.6.0
- schwifty 2024.5.2

With this command for creation:
```
pyinstaller.exe minimal.py --copy-metadata schwifty --collect-data schwifty -y --noconsole --onefile
```
And the executable is now working as expected :) => SOLVED!

Thanks again for giving this valuable infos!
